cpp/src/arrow/compute/key_hash_test.cc

```diff
@@ -311,5 +311,38 @@ TEST(VectorHash, FixedLengthTailByteSafety) {
   HashFixedLengthFrom(/*key_length=*/19, /*num_rows=*/64, /*start_row=*/63);
 }
 
+TEST(HashBatch, AllocTempStackAsNeeded) {
```

Comment:
Oh, one more thing. Do we need to test hash64 as well?

Reply:
We should add this!